test/test_reductions.py

```diff
@@ -296,8 +296,13 @@ def test_empty_tensor_empty_slice(self, device, op: ReductionOpInfo):
                 self.assertEqual(result, torch.full_like(result, torch.nan))
             else:
                 # Reducing along empty slice should raise an error
-                with self.assertRaises(IndexError):
-                    op(t, *args, dim=dim, **kwargs)
+                if isinstance(op, ReductionPythonRefInfo):
+                    # ref reductions throw RuntimeError for this
+                    with self.assertRaises(RuntimeError):
```

Comment:
If possible, we should fix the refs here. Won't block this PR on it though

Reply:
Filed issue https://github.com/pytorch/pytorch/issues/107494 to track this.